packages/react-core/src/components/FormSelect/examples/FormSelectInvalid.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { FormSelect, FormSelectOption, ValidatedOptions } from '@patternfly/react-core';
+
+export const FormSelectInvalid: React.FunctionComponent = () => {
+  const [formSelectValue, setFormSelectValue] = React.useState('');
+  const [validated, setValidated] = React.useState<ValidatedOptions>(ValidatedOptions.default);
+
+  const isEmpty = () => formSelectValue !== '';
+  const onChange = (value: string) => {
+    const validated = isEmpty() && ValidatedOptions.error;
+    setFormSelectValue(value);
+    setValidated(validated);
+  };
```

Comment:
It doesn't look like this part of the example is working how it may have been intended when originally added. 

Rather than trying to set both the validated and value states in the onChange function, it might work better to have the onChange function only call `setFormSelectValue` like other examples have, and moving the `setValidated` and checking if `formSelectValue` is an empty string to a useEffect hook that only runs when the `formSelectValue` updates. The `isEmpty` function would just also have to be updated to check that `formSelectValue === ''`

Reply:
Yes, this works better. Thanks for pointing it out.

I also renamed `const validated` variable so it doesn't collide with state variable.